genesyscloud/architect_user_prompt/genesyscloud_architect_user_prompt_proxy.go

```diff
@@ -0,0 +1,199 @@
+func (p *architectUserPromptProxy) createArchitectUserPrompt(ctx context.Context, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	return p.createArchitectUserPromptAttr(ctx, p, body)
+}
+
+// getArchitectUserPrompt retrieves a user prompt
+func (p *architectUserPromptProxy) getArchitectUserPrompt(ctx context.Context, id string, includeMediaUris bool, includeResources bool, language []string) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
+	return p.getArchitectUserPromptAttr(ctx, p, id, includeMediaUris, includeResources, language)
+}
+
+// getAllArchitectUserPrompts retrieves a list of user prompts
+func (p *architectUserPromptProxy) getAllArchitectUserPrompts(ctx context.Context, includeMediaUris bool, includeResources bool, nameArr []string) (*[]platformclientv2.Prompt, *platformclientv2.APIResponse, error, bool) {
+	return p.getAllArchitectUserPromptsAttr(ctx, p, includeMediaUris, includeResources, nameArr)
+}
+
+// updateArchitectUserPrompt updates a user prompt
+func (p *architectUserPromptProxy) updateArchitectUserPrompt(ctx context.Context, id string, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectUserPromptAttr(ctx, p, id, body)
+}
+
+// deleteArchitectUserPrompt deletes a user prompt
+func (p *architectUserPromptProxy) deleteArchitectUserPrompt(ctx context.Context, id string, allResources bool) (*platformclientv2.APIResponse, error) {
+	return p.deleteArchitectUserPromptAttr(ctx, p, id, allResources)
+}
+
+// createArchitectUserPromptResource creates a new user prompt resource
+func (p *architectUserPromptProxy) createArchitectUserPromptResource(ctx context.Context, id string, body platformclientv2.Promptassetcreate) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.createArchitectUserPromptResourceAttr(ctx, p, id, body)
+}
+
+// updateArchitectUserPromptResource updates a user prompt resource
+func (p *architectUserPromptProxy) updateArchitectUserPromptResource(ctx context.Context, id string, languageCode string, body platformclientv2.Promptasset) (*platformclientv2.Promptasset, *platformclientv2.APIResponse, error) {
+	return p.updateArchitectUserPromptResourceAttr(ctx, p, id, languageCode, body)
+}
+
+// getArchitectUserPromptIdByName retrieves a user prompt by name
+func (p *architectUserPromptProxy) getArchitectUserPromptIdByName(ctx context.Context, nameArr []string) (string, *platformclientv2.APIResponse, error, bool) {
+	return p.getArchitectUserPromptIdByNameAttr(ctx, p, nameArr)
+}
+
+func createArchitectUserPromptFn(ctx context.Context, p *architectUserPromptProxy, body platformclientv2.Prompt) (*platformclientv2.Prompt, *platformclientv2.APIResponse, error) {
+	prompt, response, err := p.architectApi.PostArchitectPrompts(body)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to create architect user prompt: %s", err)
+	}
```

Comment:
There are a few instances like this where we're adding context to the error in the proxy, and then again from the invoking function (`createUserPrompt` in this case), so the error will come out like this for the end user: `failed to create prompt 123: failed to create prompt 123: <whatever reason the api gives>` 

There are probably loads of instances in the code base where that is happening but I think we should try to move away from it and only `return nil, response, err` from the proxy. Unless of course that proxy function does more than one operation and needs to be provide more context so the user can understand what went wrong and where